uswds/uswds
# USWDS - Dependencies: POAM Dec '23

# Summary

Addressing two POAM issues, dependency updates, removing unused deps. Closes #5712.

**Before**: 51 vulnerabilities (24 moderate, 27 high).
**After:** 44 vulnerabilities (17 moderate, 27 high).

## Breaking change

This is not a breaking change.

## Related issue

Part of POAM process.

## Related pull requests

N/A

## Preview link

Preview link:
[StorybookJS preview → ](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/jm-poam-dec-23/?path=/story/design-tokens-fonts--fonts)

## Problem statement

Addressing two ongoing dependency issues with no available updates.

## Solution

1. Updated patch and minor dependencies.
2. Removed unused deps.
3. Added dependency that was used, but not in `package.json`.


## Major changes

**Removed `uswds-core`'s package.json file.** This experimental work unintentionally got published. Removing it in favor of open PR [USWDS - Packages: Prevent nested node_modules from being included in publish #5245](https://github.com/uswds/uswds/pull/5245) because its causing conflicts with project dependencies and sass unit testing (sass-embedded isn't a dependency of uswds-core).

## Testing and review

1. Run `npm install`.
2. Run `npm start`.
3. Update a SASS file and see it update in StorybookJS.
4. Running `npx gulp sassTests` or  `npm run test` should **not** fail.
5. Run gulp tasks (like build) and ensure there aren't errors and things build correctly.


## Dependency updates

| Dependency               | Old      | New      |
| :----------------------- | :------- | :------- |
| @babel/core              | 7.23.2   | 7.23.6   |
| @babel/preset-env        | 7.23.2   | 7.23.6   |
| @types/node              | 20.8.9   | 20.10.4  |
| eslint                   | 8.52.0   | 8.55.0   |
| eslint-config-prettier   | 9.0.0    | 9.1.0    |
| handlebars-helpers  [^1] | 0.10.0   | -        |
| html-webpack-plugin      | 5.5.3    | 5.5.4    |
| postcss                  | 8.4.31   | 8.4.32   |
| postcss-import      [^2] | -        | 15.1.0   |
| postcss-preset-env       | 9.2.0    | 9.3.0    |
| postcss-sass-loader [^3] | 1.1.0    | -        |
| sass [^4] |  - |  1.69.5 |
| snyk                     | 1.1237.0 | 1.1262.0 |
| svgo                     | 3.0.2    | 3.1.0    |
| typescript               | 5.2.2    | 5.3.3    |


[^1]: Previously used for token JSON to SASS conversion.
[^2]: Was already being used, but not included directly in package.json.
[^3]: Dependency that wasn't called directly. Hasn't been updated in 5 years now.
[^4]: Added as a dependency for `sass-true`.

Also created #5674 to address an outstanding nanoid issue.
